### PR TITLE
[Core ML delegate] Forward declaration

### PR DESCRIPTION
**Pull request by idstein**
Add a forward declaration to TfLiteDelegate

It's faster compile time and reduced binary size besides avoiding namespace pollution without including "tensorflow/lite/c/common.h"

### CODE REVIEW

**Comment by MeghnaNatraj** [comment]
@srjoglekar246 could you take a look at this? Thank you!

**Comment by teijeong** [comment]
Hi idstein,

Can you add the reason of this change in the commit description?

File: tensorflow/lite/experimental/delegates/coreml/coreml_delegate.h [hunk 1]
```diff
@@ -15,11 +15,12 @@ limitations under the License.
 #ifndef TENSORFLOW_LITE_EXPERIMENTAL_DELEGATES_COREML_COREML_DELEGATE_H_
 #define TENSORFLOW_LITE_EXPERIMENTAL_DELEGATES_COREML_COREML_DELEGATE_H_
 
-#include "tensorflow/lite/c/common.h"
```
Comment (terryheo) [inline tensorflow/lite/experimental/delegates/coreml/coreml_delegate.h:18]: what's the benefit of removing this?

**Comment by gbaned** [comment]
@idstein  Can you please check @terryheo's comments and keep us posted ? Thanks!

**Comment by gbaned** [comment]
@idstein  Any update on this PR? Please. Thanks!

**Comment by tensorflowbutler** [comment]
It has been 14 days with no activity and the `awaiting response` label was assigned. Is this PR still valid? Assigning the `stalled` label. Please comment to reassure me that this is still being worked on.

**Comment by idstein** [comment]
As only a single symbol is referenced adding the entire common header seems to pollute a lot the namespaces.

When removed it improves compilation time and binary size besides the aforementioned namespace reduction.

File: tensorflow/lite/experimental/delegates/coreml/coreml_delegate.h (hunk 1 above)
Comment (idstein) [reply tensorflow/lite/experimental/delegates/coreml/coreml_delegate.h:18]: The benefit is faster compile time and reduced binary size besides avoiding namespace pollution.

Comment (terryheo) [reply tensorflow/lite/experimental/delegates/coreml/coreml_delegate.h:18]: I don't see the improvement on my side. What's your test method?

**Comment by gbaned** [comment]
@terryheo  Can you please take a look on the above comment from @idstein. Thanks!

**Comment by terryheo** [comment]
@idstein , could you share the way you used to measure the improvement?
I'm reluctant to accept this since "tensorflow/lite/c/common.h" is basic header so eventually we might need to include this again even though we don't this it with the forward declaration at this time.

So I'd like to see how much compile time benefit we can get from this. In my testing, there is no significant difference in building time.

**Comment by gbaned** [comment]
@idstein  Can you please check @terryheo's comments and keep us posted ? Thanks!

**Comment by gbaned** [comment]
@idstein Any update on this CL? Please. Thanks!

**Comment by idstein** [comment]
PR can be closed as there have been new dependencies introduced in current master. Sorry for the late response.

**Pull request closed without merging**